apps/users-service/src/address-book-service.service.ts

```diff
@@ -107,4 +113,33 @@ export class AddressBookService {
 
     return { status: 1 }
   }
+
+  async getAddressByPin (userId: string, pin: number): Promise<PinAddressI> {
+    try {
+      const getUser: UserI = await this.usersRepository.findOne({
```

Comment:
Only get the user with the PIN no userID. Remember the user hitting this endpoint is not the user with the address.